### PR TITLE
feat: add Organization Invoices Search API

### PR DESCRIPTION
**Pull request by whoAbhishekSah**
# Add Organization Invoices Search API

This PR adds a new endpoint for searching organization invoices, with support for filtering, sorting, searching, and grouping capabilities.

## API Details
```http
POST /v1beta1/admin/organizations/{org_id}/invoices/search
```

### Features
- **Pagination** with offset and limit
- **Filtering** on amount, state, currency, and created_at
- **Sorting** on state, amount, and created_at fields
- **Text search** across state, hosted_url, and amount fields
- **Group By** support for state field, returning counts per state

### Example Request
```json
{
    "filters": [
        {
            "name": "amount",
            "operator": "gte",
            "number_value": 0
        }
    ],
    "sort": [
        {
            "name": "amount",
            "order": "desc"
        }
    ],
    "search": "test",
    "group_by": ["state"],
    "offset": 0,
    "limit": 100
}
```

### Supported Operations
- **Filters**:
  - Numeric operators: eq, neq, gt, lt, gte, lte
  - String operators: eq, neq, like, in, notin, notlike, empty, notempty
  - Date operators: eq, neq, gt, lt, gte, lte

- **Sort Fields**: state, amount, created_at
- **Group By**: state (returns count of invoices per state)
- **Search**: Performs text search on state, hosted_url, and amount fields

## Implementation Details
- Uses RQL (REST Query Language) for flexible querying
- Uses prepared statements for security
- Maintains transaction consistency
- Includes comprehensive unit tests

## Testing
- Added unit tests for query preparation
- Tested various combinations of filters, sort, search, and group by
- Verified pagination functionality
- Tested error cases and edge conditions

Related PR: https://github.com/raystack/proton/pull/397

### CODE REVIEW

File: core/aggregates/orginvoices/service.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	Offset int `json:"offset"`
+}
+
+type AggregatedInvoice struct {
+	ID          string    `rql:"name=id,type=string"`
+	Amount      int64     `rql:"name=amount,type=number"`
+	Currency    string    `rql:"name=currency,type=string"`
+	State       string    `rql:"name=state,type=string"`
+	InvoiceLink string    `rql:"name=invoice_link,type=string"`
+	BilledOn    time.Time `rql:"name=billed_on,type=datetime"`
```
Comment (rsbh) [inline core/aggregates/orginvoices/service.go:51]: use only `created_at` instead of `billed_on`

**Review by rsbh — APPROVED** [review]
